pages/myDonations.js

```diff
@@ -0,0 +1,40 @@
+import styled from "styled-components";
+import MyDonationList from "../components/MyDonationList";
+import AddDonation from "@/components/AddDonation";
+import Link from "next/link";
+
+export default function MyDonations({
+  donations,
+  createDonation,
+  handleDelete,
+  handleDummyDonations,
+}) {
+  return (
+    <>
+      <h2> Meine Spenden:</h2>
+      <Link href={"/"}>back</Link>
+      <br />
+      <button type="button" onClick={handleDummyDonations}>
+        Reset to default
+      </button>
+      <AddDonation createDonation={createDonation} />
+
+      <StyledCard>
+        {donations.map((donation) => {
+          return (
+            <MyDonationList
+              key={donation.id}
+              donation={donation}
+              handleDelete={handleDelete}
+            />
```

Comment:
is it possible that you create exactly the same cards here as you did on your homepage, only instead of listing all donations only mine are displayed? If that is true, you should use the same Card component for the job, and display the delete button if the donation was created by me. You can check that by adding a key to each donation Object that is called createdBy. The value is the userID of the creator. For now you don't have a dedicated userID so you can mock them up. In your MyDonation Page you need to define a "currentUserID" which defines which cards are displayed. Then you can filter the donations that only donations with matching createdBy userID are rendered. This filtering must be done before you map over the donations array. If this was a bit too cryptic you can ask me for assistance.

Reply:
you are right- added a UserID and a filter function- thx 🦦